.github/pull_request_template.md

```diff
@@ -1,66 +1,51 @@
-- [ ] This PR adds/removes an experimental interfaces and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
-- [ ] I tagged two reviewers because this PR makes queries to the database or I think it might have some security impact.
-
-### Screenshots
-
-<!-- If your change has a visual component, add a screenshot here. A "before" screenshot would also be helpful. -->
+<!-- Any screenshots of UI changes will be helpful to include here -->
 
 | Before | After |
 | ------ | ----- |
 |        |       |
 
-### Testing
-
-#### Automated Tests
-* [ ] Changes in this PR are covered by Automated Tests.
-  * [ ] Unit tests
-  * [ ] E2E tests
-
-#### User Facing Testing
+## Testing Instructions
 
 <!-- Write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->
 
+_Please consider any edge cases this change may have, and also other areas of the product this may impact._
+
 1.
 2.
 3.
 
 * [ ] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested (ie: it makes changes to tests, coding standards, docblocks, etc.). -->
 * [ ] Should be tested by the development team exclusively <!-- Check this box if this PR should be tested by the development team exclusively (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->
 
-### WooCommerce Visibility
+## WooCommerce Visibility
 
 <!-- Check this [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WC core or not (part of the feature plugin or experimental)-->
 
 * [ ] WooCommerce Core
 * [ ] Feature plugin
 * [ ] Experimental
 
-### Performance Impact
+## Checklist
 
-<!-- Please document any known performance impact (positive or negative) here. If negative, provide some rationale for why this is an okay tradeoff or how this will be addressed. -->
+* [ ] This PR has either a `[type]` label or a `[skip-changelog]` label.
+* [ ] This PR is assigned to a milestone.
+* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
+* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
+* [ ] This PR has been accessibility tested.
+* [ ] This PR has a changelog description.
```

Comment:
One suggestion I'd like to propose is regarding the **use of checkboxes** in the template:

Currently, the checkboxes serve multiple purposes – some seem to be absolutely required (like assigning the PR to a milestone or adding a label), while others are conditionally relevant (like updating a feature flag or an experimental interface).

To make the expectations clearer for anyone raising a PR, I suggest we categorize the checkboxes into two distinct sections:

1. **Required Actions**: Checkboxes under this category should represent tasks that are mandatory for every PR. For instance, adding a label, assigning a milestone. This ensures essential steps are not missed.
2. **Conditional Actions**: These checkboxes would apply if the PR involves certain types of changes. For example, updating a feature flag or an experimental interface would fall under this category. If a PR doesn't involve such changes, it's perfectly fine for these boxes to remain unchecked.

Reply:
You're absolutely right @imanish003 and great suggestion!